metrics/prometheus.go

```diff
@@ -57,30 +57,30 @@ func (p *PromMetrics) Register(name string, metricType string) {
 	switch metricType {
 	case "counter":
 		newmet = promauto.NewCounter(prometheus.CounterOpts{
-			Name: name,
+			Name:      name,
 			Namespace: p.prefix,
-			Help: name,
+			Help:      name,
 		})
 	case "gauge":
 		newmet = promauto.NewGauge(prometheus.GaugeOpts{
-			Name: name,
+			Name:      name,
 			Namespace: p.prefix,
-			Help: name,
+			Help:      name,
 		})
 	case "histogram":
 		newmet = promauto.NewHistogram(prometheus.HistogramOpts{
-			Name: name,
+			Name:      name,
 			Namespace: p.prefix,
-			Help: name,
+			Help:      name,
 		})
 	}
 
 	p.metrics[name] = newmet
 }
 
 func (p *PromMetrics) Increment(name string) {
-	p.lock.Lock()
-	defer p.lock.Unlock()
+	p.lock.RLock()
+	defer p.lock.RUnlock()
```

Comment:
Does Increment() only need a read lock?

Reply:
That's correct per https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#section-readme

> All exported functions and methods are safe to be used concurrently unless specified otherwise.

Reply:
Bueno.